### ISSUE TITLE
Use PyFileSystem for Memory-based AppSource

### ISSUE THREAD

**Issue by gregelin** [GovReady/govready-q#682]
As a developer, I want to treat Memory as a filesystem so I can define an AppSource that can be used for temporary files in a readonly environment. 

*This will also fix a bandit issue in CircleCI testing related to using the /tmp directory.*

Our solution will be to use PyFileSystem2, which we are already requiring, to create a file system in RAM. See https://www.pyfilesystem.org/page/memoryfs/ .

The fix should involve very little code. We just need to set up the Memoryfs to use each time. We need the data written to the memoryfs to persist for only the duration of our processing.

We want to fix in in `first_run`:
```
        # Finally, for authoring
        # Q currently only works on Unix systems; this is OK bc Window users have to use Docker version locally
        qfiles_path = "/tmp"
        if os.path.exists(qfiles_path):
            AppSource.objects.get_or_create(
                slug="tmp",
                defaults={
                    "spec": { "type": "local", "path": qfiles_path }
                }
            )
```

We want to fix local or in memory file systems in `guidedmodules.views.authoring_create_q`:
```
    # What's the path to the app?
    path = os.path.join(new_q_appsrc.spec["path"], new_q["q_slug"])

    # Does a Q already exist with directory name?
    # TODO: Better test, test name in database and path?
    if os.path.exists(path):
        print("A project with this slug already exists {}".format(os.path.exists(path)))
        return

    # Copy stub files.
    guidedmodules_path = os.path.dirname(__file__)

    # Write temporary file
    shutil.copytree(os.path.join(guidedmodules_path, "stub_app"), path, copy_function=shutil.copy)

    # Edit the app title and catalog information.
    with rtyaml.edit(os.path.join(path, "app.yaml")) as app:
        app['title'] = new_q["title"]
        app['catalog']['description']['short'] = new_q['short_description']
        app['introduction']['template'] = new_q['short_description']
        app['category'] = new_q['category']

    # Create a unique icon for the app and delete the existing app icon
    # svg file that we know is in the stub.
    from mondrianish import generate_image
    colors = ("#FFF8F0", "#FCAA67", "#7DB7C0", "#932b25", "#498B57")
    with open(os.path.join(path, "assets", "app.png"), "wb") as f:
        generate_image("png", (128, 128), 3, colors, f)
    # Clean up
    os.unlink(os.path.join(path, "assets", "app.svg"))

    # Publish our new app
    try:
        appver = new_q_appsrc.add_app_to_catalog(new_q["q_slug"])
    except Exception as e:
        raise
```

We want to improve to be conditional based on where we are running in `compliance_app.py`:
```
            # Copy stub files.
            guidedmodules_path = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
            shutil.copytree(os.path.join(guidedmodules_path, "stub_app"), path, copy_function=shutil.copy)
```

**Comment by gregelin** [comment]
Another strategy might be to create an appsource for the stub app and simply publish the stub app into the catalog and then begin to edit the stub app.

**Comment by gregelin** [comment]
The second strategy of creating the new app from the stub app worked much better.

Issue fixed with commit c8e280adc1c51c23c0b236060613aff6febe94a9
